validator/patch.go

```diff
@@ -56,39 +62,36 @@ func GetPatchedProject(ctx context.Context, p *patch.Patch, githubOauthToken str
 		}
 	}
 
-	project := &model.Project{}
-
-	// if the patched config exists, use that as the project file bytes.
-	if p.PatchedConfig != "" {
-		projectFileBytes = []byte(p.PatchedConfig)
-	}
-
 	// apply remote configuration patch if needed
-	if !p.IsGithubPRPatch() && p.ConfigChanged(projectRef.RemotePath) && p.PatchedConfig == "" {
-		project, err = model.MakePatchedConfig(ctx, p, projectRef.RemotePath, string(projectFileBytes))
+	if !(p.IsGithubPRPatch() || p.IsPRMergePatch()) && p.ConfigChanged(projectRef.RemotePath) {
+		projectFileBytes, err = model.MakePatchedConfig(ctx, p, projectRef.RemotePath, string(projectFileBytes))
 		if err != nil {
 			return nil, errors.Wrapf(err, "Could not patch remote configuration file")
 		}
-		// overwrite project fields with the project ref to disallow tracking a
-		// different project or doing other crazy things via config patches
-		var verrs ValidationErrors
-		verrs, err = CheckProjectSyntax(project)
-		if err != nil {
-			return nil, errors.WithStack(err)
-		}
+	}
 
-		if len(verrs) != 0 {
-			var message string
-			for _, err := range verrs {
-				message += fmt.Sprintf("\n\t=> %+v", err)
-			}
-			return nil, errors.New(message)
-		}
-	} else {
-		// configuration is not patched
-		if err = model.LoadProjectInto(projectFileBytes, projectRef.Identifier, project); err != nil {
-			return nil, errors.WithStack(err)
+	return projectFileBytes, nil
+}
+
+func ValidateProjectPatch(data []byte, projectID string) (*model.Project, bool, error) {
```

Comment:
This should project just take a *model.Project and return a bool

Reply:
This function was removed.